src/services/mongoose/wallet-on-chain-addresses.ts

```diff
@@ -0,0 +1,61 @@
+import { Types as MongooseTypes } from "mongoose"
+import { User } from "@services/mongoose/schema"
+import {
+  CouldNotFindError,
+  RepositoryError,
+  UnknownRepositoryError,
+} from "@domain/errors"
+
+export const WalletOnChainAddressesRepository = (): IWalletOnChainAddressesRepository => {
+  const persistNew = async (
+    walletId: WalletId,
+    onChainAddress: OnChainAddressIdentifier,
+  ): Promise<OnChainAddressIdentifier | RepositoryError> => {
+    try {
+      const { address, pubkey } = onChainAddress
+      const result = await User.updateOne(
+        { _id: walletId },
+        { $push: { onchain: { address, pubkey } } },
+      )
+
+      if (result.n === 0) {
+        return new CouldNotFindError("Couldn't find wallet")
+      }
+
+      if (result.nModified !== 1) {
+        return new RepositoryError("Couldn't add onchain address for wallet")
```

Comment:
So far `RepositoryError` is only used as a base marker and never created directly. Perhaps this could be a `PersistError` or something like that.

Reply:
fixed